include/itkANTSRegistration.hxx

```diff
@@ -129,20 +173,122 @@ ANTSRegistration<TFixedImage, TMovingImage, TParametersValueType>::AllocateOutpu
 }
 
 
-//template <typename TFixedImage, typename TMovingImage, typename TParametersValueType>
-//auto
-//ANTSRegistration<TFixedImage, TMovingImage, TParametersValueType>::MakeOutput(DataObjectPointerArraySizeType)
-//  -> DataObjectPointer
+// template <typename TFixedImage, typename TMovingImage, typename TParametersValueType>
+// auto
+// ANTSRegistration<TFixedImage, TMovingImage, TParametersValueType>::MakeOutput(DataObjectPointerArraySizeType)
+//   -> DataObjectPointer
 //{
-//  return DataObjectPointer();
-//}
+//   return DataObjectPointer();
+// }
 
 
+template <typename TFixedImage, typename TMovingImage, typename TParametersValueType>
+template <typename TImage>
+auto
+itk::ANTSRegistration<TFixedImage, TMovingImage, TParametersValueType>::CastImageToInternalType(
+  const TImage * inputImage) -> typename InternalImageType::Pointer
+{
+  using CastFilterType = CastImageFilter<TImage, InternalImageType>;
+  typename CastFilterType::Pointer castFilter = CastFilterType::New();
+  castFilter->SetInput(inputImage);
+  castFilter->Update();
+  typename InternalImageType::Pointer outputImage = castFilter->GetOutput();
+  outputImage->DisconnectPipeline();
+  return outputImage;
+}
+
 template <typename TFixedImage, typename TMovingImage, typename TParametersValueType>
 void
 ANTSRegistration<TFixedImage, TMovingImage, TParametersValueType>::GenerateData()
 {
-  this->AllocateOutputs();
+  // this->AllocateOutputs();
+
+  std::stringstream ss;
+  m_Helper->SetLogStream(ss);
+
+  std::string whichTransform = this->GetTypeOfTransform();
```

Comment:
-> behavior of ANTs with possibly multiple transforms

Reply:
We later agreed to do one transform at a time. If multiple stages are needed, solve it via setting the initial transform.

Reply:
To be clear, we discussed following the behavior of [antspyx' python interface](https://antspyx.readthedocs.io/en/latest/registration.html), which has some pre-defined multi-transform optimizations, e.g. `"TRSAA"`, which are registered in call. But additional, multi-stage registrations would be made via multiple calls.